neurolib/models/multimodel/builder/base/backend.py

```diff
@@ -184,16 +206,67 @@ def _substitute_helpers(self, derivatives, helpers):
         substitutions = {helper[0]: helper[1] for helper in sympified_helpers}
         return [derivative.subs(substitutions) for derivative in sympified_derivatives]
 
-    def run(self, duration, dt, noise_input, **kwargs):
+    @staticmethod
+    def _get_numba_function_params(symbol_params):
         """
-        Run integration.
+        Get all symbolic parameters as a list for numba function string template.
 
-        :kwargs: actually none - for compatiblity
+        :param symbol_params: symbolic parameters as a flat dict
+        :type symbol_params: dict
+        :return: list of all parameters as symbols
+        :rtype: list[sp.Symbol]
         """
-        assert isinstance(noise_input, np.ndarray)
-        system_size = len(self._derivatives())
+        all_params = set()
+        for k, v in symbol_params.items():
+            if "noise" in k:
+                continue
+            if isinstance(v, sp.Symbol):
+                # extract symbols from sympy expressions
+                all_params |= v.free_symbols
+            elif isinstance(v, np.ndarray):
+                # extract symbols from sympy expressions and do the union of all sets
+                all_params |= set().union(*[item.free_symbols for item in v.flatten().tolist()])
+        return list(all_params)
+
+    @staticmethod
+    def _create_symbol_to_float_dict(symbol_params, float_params):
```

Comment:
this function builds the dictionary of mapping {`symbol parameter`: `its float value`}